### PR TITLE
fix connmgr races

### PR DESCRIPTION
**Pull request by maxtaco**
You can only access the lookup from inside the inner loop. public meethods should not penetrate the interior.

### CODE REVIEW

**Comment by maxtaco** [comment]
r? @mmaxim 

CC: @gabriel

File: go/libkb/connmgr.go [hunk 1]
```diff
@@ -48,13 +53,14 @@ type ConnectionManager struct {
...
-	labelConnectionCh  chan labelConnectionObj
-	listAllCh          chan chan<- []keybase1.ClientDetails
+	addConnectionCh      chan *addConnectionObj
+	lookupConnectionCh   chan *lookupConnectionObj
+	removeConnectionCh   chan ConnectionID
+	applyAllCh           chan ApplyFn
+	shutdownCh           chan struct{}
+	labelConnectionCh    chan labelConnectionObj
+	listAllCh            chan chan<- []keybase1.ClientDetails
+	lookupByClientTypeCh chan *lookupByClientTypeObj
```
Comment (jzila) [inline go/libkb/connmgr.go]: Need to initialize this in `NewConnectionManager()`

**Comment by maxtaco** [comment]
I don't see a way for v to be nil

I think it was nil as a result of a race.  What do you guys think?

**Comment by maxtaco** [comment]
thanks for catching it

apparently this code isn't tested in client CI

**Pull request merged**